### ISSUE TITLE
AI Units in basic medical are not dying after first (few?) AI deaths

### ISSUE THREAD

**Issue by Knight1981** [acemod/ACE3#1155]
ACE3 Version: 3.0.0

**Mods:**
    @cba_a3
    @ace3

**Placed ACE3 Modules:**
MicroDAGR MapFill (standard values)
MK6 Settings (standard values)
NameTags (standard values)
Weather (standard values)
Wind Deflection (standard values)
Explosive System (standard values)
Hearing (standard values)
Interaction system (standard values)
Map (standard values)
Medical Settings (Level = Basic, Setting = Disable Medics, AI uncons. = disabled, prevent instant death = yes)
Set Medic Class (we just defined 3 medics as regular medic)
Basic Revive (disabled)

**Description:**
We have the problem that some AI don't die! We can shot them with 100 rounds but nothing happened on many AI units. We are using the basic medical system which is disabled for AI. 

**Steps to reproduce:**
Create a mission with ace and basic medical system (not advanced!). Set the "prevent instant death = yes". Place some enemies and try to shot them down on a dedicated server (not locally).

**Where did the issue occur?***
Multiplayer, playing on dedicated Server

**RPT log file:**
http://www.krisenkommandokraefte.de/temp/arma3_2015-05-14_23-31-43.rpt

**Sample Mission file:**
http://www.krisenkommandokraefte.de/temp/mission.sqm

**Comment by jaynus** [comment]
Please edit your report to follow the guidelines: http://ace3mod.com/wiki/user/how-to-report-an-issue.html

**Comment by Desty84** [comment]
Happens also in Advanced!
And it looks like a compatibility problem with RHS.
I am using RHS + HLC weapon mods, the RHS weapons just can´t kill the AI but the HLC weapons can. (with 20-30 shots, but they die)

I tried to change the Damage Values in the Basic module, but with a value under 1 they are getting invincible.

**Comment by Knight1981** [comment]
@Desty84: Not only with RHS. We are using the G36 of the BW-Mod and having the same problem. First we had the AI damage set to 1 and we also tried it with 0.6.

#############################
ACE3 Version: 3.0.0

**Mods:**
* @cba_a3
* @ace3
* @bwa3
* @Alive
* @mcc_sandbox
* @ASDG_JR
* @CAF_AG

**Placed ACE3 Modules:**
MicroDAGR MapFill (standard values)
MK6 Settings (standard values)
NameTags (standard values)
Weather (standard values)
Wind Deflection (standard values)
Explosive System (standard values)
Hearing (standard values)
Interaction system (standard values)
Map (standard values)
Medical Settings (Level = Basic, Setting = Disable Medics, AI uncons. = disabled, prevent instant death = yes)
Set Medic Class (we just defined 3 medics as regular medic)
Basic Revive (disabled)


**Description:**
See above (1st topic). We test this with different fractions (CAF, Civilians, Opfor) with the BW-Mod weapon (G36).

**Steps to reproduce:**
Place a lot of enemies and try to shot them down :D Some of the enemies where placed by MCC or Alive, some manually.

**Where did the issue occur?**
Multiplayer, playing on dedicated Server

**RPT log file:**
do not have, sorry

**Comment by Glowbal** [comment]
This appears to be two issues:

1) prevent instant death
2) Damage threshold for basic medical

**Comment by Desty84** [comment]
Got a workaround for this issue!

If you set "prevent insta death" to disabled and place the revive module with the settings:

Enable Revive: Players only
Max Revive Time: >0 (for me 600)
Max Revive Lifes: -1

it works pretty good.
AI dies normally, players get unconscious but don´t die instant.

Hope that helps!

**Comment by Vizharan** [comment]
@Desty84 
Can you provide me a link with a working "test" mission? I also have these settings, but my a.i. is almost invulnerable.

The culprit is the "prevent instant death" setting.

**Comment by Desty84** [comment]
@Vizharan

Try to copy the modules to your mission and test it!
For me it works, tested it with a heavily modded I&A. 
(The modeset is very big so i hope the modules are enough for your purpose)

https://www.dropbox.com/s/cj5d4ohyj6yu6k2/ace_medsys_test.Altis.zip?dl=0

**Comment by Vizharan** [comment]
@Desty84 

Thanks for the setup.

Looks like you need the "Advanced Medical Module" AND Advanced Medical Settings. Only with these two enabled the Revive module will work.

**Comment by Glowbal** [comment]
What is the issue with the revive module in basic?

**Comment by headedhorseman101** [comment]
Had a similar issue with agm in the exact same setting (of course this is agm, but thought that it might be of help).

**Comment by Knight1981** [comment]
@Desty84 
Thank you for the workaround, but it doesn't work :(

We disabled the "prevent instant death" and activated the basic revive module with these settings:
Enable Revive: Players only
Max Revive Time: >0 (for me 600)
Max Revive Lifes: -1

In some cases I was unconscious but on critical hits I died instantly. So this is not an option to disable "prevent instant death".

This bug is really critical for us at the moment because we can't play any coop mission with these settings :( Prevent instant death is very important for us, we won't that some of us is dying instantly.

**Comment by Glowbal** [comment]
I cannot reproduce. Please confirm this also happens with just CBA and ACE3.

**Comment by Glowbal** [comment]
Also please provide rtp log files.

**Comment by Knight1981** [comment]
Here the rtp: http://www.krisenkommandokraefte.de/temp/arma3_2015-05-14_19-30-27.rpt

To reproduce please set the modules and settings as I described in first post. Some AI dying immediately and some don't die after 100 shots.

**Comment by jonpas** [comment]
> == "C:\Program Files (x86)\Steam\SteamApps\common\Arma 3\arma3.exe" -nosplash -world=empty -skipIntro -maxMem=4094 -enableHT -high -mod=@DAR;@task_force_radio;@Sangin;@RDS;@kyo_MH47E;@blastcore-A3;@A3MP;@ASDG_JR;@BWA3;@CAF_AG;@ALiVE;@BW_Fenneks;@mcc_sandbox;@HiddenIdentityPackV2;@RAV_Lifter_A3;@bw_tornado;@RQ-11_RAVEN_AB_A3;@L_Mount;@L_Climb;@NATO_Hellcats;@akp_spotlight;@DragonFyre;@Helo_HMDs_Kimi;@US_helos_Kimi;@A3MP-AP;@fata;@BWDR;@AWC_bags;@FEMAL3;@Clafghan;@CAF_AG_AudioPatch;@CBA_A3;@BW_Kleiderkammer;@ace;

Please test using only `@cba_a3` and `ace3`.

**Comment by alganthe** [comment]
OH MY GOD no fucking wonder you're having issues ...

**Comment by Glowbal** [comment]
> To reproduce please set the modules and settings as I described in first post. Some AI dying immediately and some don't die after 100 shots.

I did. I cannot reproduce. Please try with only `@CBA` and `@ace`.

**Comment by Knight1981** [comment]
@Glowbal: Yes! The error orrurs also when I'm starting only @CBA and @ACE, but only on dedicated server. When I'm click on preview in the editor it works fine and the AI all dies. So please test this on a dedicated server!

I have created a small test mission for you with 1 player and some enemys without weapons. Try to test this mission on your dedicated server and kill all AI's. Some of them are dying, some of them not. In the mission you can see my modules.

http://www.krisenkommandokraefte.de/temp/mission.sqm

Perhaps you have to test this more than 1 time.

**Comment by krachenko** [comment]
Same problem here, with basic module  @Knight1981@Glowbal

**Comment by ESUSSanchez** [comment]
Problem only in dedicated. Im loading with:

@ace
@ALiVE
@ASDG_JR
@ASR_AI3
@CBA_A3
@Leights_OPFOR
@PRKZ
@RHSAFRF
@RHSUSAF
@RH_acc
@task_force_radio

First IA i shoot dies but next its invincible, and next and next...
This happens with basic and advanced settings. 

Thanks!!!

**Comment by jonpas** [comment]
Ok, something is apparently not clear here, you have to test with ONLY `@cba_a3` and `@ace` and ONLY them, only then we can be sure it's ACE's fault.

@Knight1981 The RPT you posted shows you are loading a bunch of mod, as I posted in my above comment, here they are again:
`@DAR;@task_force_radio;@Sangin;@RDS;@kyo_MH47E;@blastcore-A3;@A3MP;@ASDG_JR;@BWA3;@CAF_AG;@ALiVE;@BW_Fenneks;@mcc_sandbox;@HiddenIdentityPackV2;@RAV_Lifter_A3;@bw_tornado;@RQ-11_RAVEN_AB_A3;@L_Mount;@L_Climb;@NATO_Hellcats;@akp_spotlight;@DragonFyre;@Helo_HMDs_Kimi;@US_helos_Kimi;@A3MP-AP;@fata;@BWDR;@AWC_bags;@FEMAL3;@Clafghan;@CAF_AG_AudioPatch;@CBA_A3;@BW_Kleiderkammer;@ace;`
Please test using ONLY `@cba_a3` and `@ace`, see if you can reproduce it and post an RPT from that session.

@ESUSSanchez Please test using ONLY `@cba_a3` and `@ace` and see if you can reproduce it there as well, pretty sure the problem lies in some other mod changing something it shouldn't.

@krachenko RPT please.

**Comment by ESUSSanchez** [comment]
@jonpas Hi again here example mission with same modules as in original. 

https://mega.co.nz/#!uNpD1KrR!NgGzb2a576M3CwAuVMXmHKCKFdtcOeZBBcG1uZ67Xx8

Here a video testing on dedicated server with 2 clients. 

https://mega.co.nz/#!yYRDkBxZ!YZBHPDSBnz4qvMCXsRLIFbKJTHRgHjX-fu7UaD4dysI

Here server RPT

https://mega.co.nz/#!eIhzgJ5I!nbzLsAOgl6twHsgtz0_HrRejXW_5ifhu1kZVLZdqDQY

**Comment by krachenko** [comment]
@jonpas  I am playing with @ESUSSanchez in the same server.
Its rpt  applies to my previous answer

**Comment by Knight1981** [comment]
@jonpas: Yes, the first rpt file shows more mods, but as I wrote later I have tested this only with CBA and ACE. Please check my sample mission, there is only CBA and ACE loaded and the problem still occur. So I'm 100% sure that this is an ace problem with basic medical module and "prevent instant death = yes". No other mods were loaded in my 2nd test!

**Comment by alganthe** [comment]
welp that's a non issue prevent instant death apply to players AND ai

**Comment by ramius86** [comment]
Same issue, with a personal build I did 1 day before official release all was working fine: with official release and same mod and mission ( nothing changed at all ) got this bug

**Comment by Glowbal** [comment]
Could someone record this and upload a video of the bug happening?

**Comment by krachenko** [comment]
@glowbal this video is from @ESUSSanchez and I of the issue happening, https://mega.co.nz/#!yYRDkBxZ!YZBHPDSBnz4qvMCXsRLIFbKJTHRgHjX-fu7UaD4dysI

**Comment by ramius86** [comment]
@krachenko mine bug is same but a little different: AI die after like 100-150 bullets

**Comment by Glowbal** [comment]
Could you try with setting the AI damage threshold to 1 (the default value) instead of 0.5?

**Comment by Knight1981** [comment]
@Glowbal: I already wrote above that we tested this with 1 and 0.6.

**Comment by headedhorseman101** [comment]
I had the same problem @Knight1981 described in both ACE 3 and agm (using a backlogged version of CBA to run agm, I think it is R5 with hotfix and latest CBA to test with ACE) . If you are using similar scripting commands for the setting, I would watch the lines that are the same in agm and ace while debugging ACE.

**Comment by Grollig** [comment]
Can confirm there is a problem with the basic medical system and prevent instant death set to 'yes'. Some AI units simply were immune to bullets, while others died normally. I have just uploaded a mission to our dedicated server and the problem occured only there. Local testing was all fine.

**Comment by jaynus** [comment]
Can anyone clarify please on what they mean by 'immune'. As in, they never go unconcious? Or they never die? Or you waited 10 minutes to see if they bled out and they never did?

Do they get back up and walk again? Or is it you are just not getting the 'kill' in the stats?

**Comment by Grollig** [comment]
In the mission we just tested (first live test of ace3 with basic medic system enabled on our dedicated server) we had two enemies right in front of us. The first one died after two almost simultaneous hits in his chest. Felt pretty normal. The second guy however took several shots all over his body and just kept standing there, staring at us with his rifle in lowered position. He didn't walk, but he could still turn, as we gathered around him. After we shot several magazines at him, he finally died. Can't describe it better, sorry.

**Comment by Nagrach** [comment]
We noticed the same "immune" problem.
AI taking hits and starts bleeding but wont fall unconscious nor die due a "normal" amount of bullets.
We tested it with Zeus and dopped some AI and noticed that all those "immune" ai stopped losing health at a value around 10% (Zeus AI life/armor bar)

If we fill up the health of a unit manually via zeus - the next hit kills the unit.
If we lower the health manually to something like 10% - the unit starts to be immune to future dmg until a "critical hit" kills it. 

Testet with CBA (RC6 ) Ace3 and Tfar

**Comment by Jester814** [comment]
jaynus - they are immune. They seem to be taking some initial damage, but once they get to a certain point they simply stop taking damage. Regardless of what they're hit with. Bullets, frags, 40mm grenades, cannon rounds, mortars, etc. Complete immunity to damage.

**Comment by jaynus** [comment]
@glowbal this seems a bit more serious and is definitely reproducing across the board.

**Comment by Jester814** [comment]
I've got a video coming with some thoughts. Standby ~1 hr

**Comment by ramius86** [comment]
I've rolled back to a build I did 12th may around 22-23pm GMT+1 and all is working fine so if you want to catch the bug check what is changed between :+1:

**Comment by Jester814** [comment]
Video: https://www.youtube.com/watch?v=fh44X6LURk4

RPT from that session: https://drive.google.com/file/d/0B0Q6muNWoUOIYnZvMm1LbHNXbnM/view?usp=sharing

Note: My RPT isn't doing timestamps anymore and the server was on for a very long time. Look from the bottom up. I'm not sure when I connected and started the mission fresh, but the end of the RPT is just as we concluded testing.

Possible points of interest:

-Doesn't seem to affect people that were connected to the mission when the mission started. People that connect after mission start often start experiencing the bug instantly.

-Some people will experience it while others won't. Different people shooting the same AI can have different results

-Seems to happen when the AI is low on health and the next shot should/would kill him

-This was happening with much greater frequency in prior testing

Hope this helps. This is definitely a critical bug for us.
